lib/utilities/apiWrapper.ts

```diff
@@ -0,0 +1,57 @@
+
+import fetch from './fetch'
+
+import { BUTTER_BASE_API_URL, BUTTER_BASE_HEADERS } from '../config'
+import type { GlobalButterConfig } from '../typescript/GlobalButterConfig'
+
+export class APIWrapper {
+  #token: string;
+  #previewMode: boolean;
+  #timeout: number;
+  #config: GlobalButterConfig
+  #baseURL = BUTTER_BASE_API_URL;
+
+  constructor (token: string, previewMode: boolean = false, timeout: number = 3000, config: GlobalButterConfig) {
+    this.#token = token
+    this.#previewMode = previewMode
+    this.#timeout = timeout
+    this.#config = config
+  }
+
+  // rome-ignore lint/suspicious/noExplicitAny: <explanation>
+  createParams (_params?: Record<string, any>) {
+    const params = {
+      auth_token: this.#token,
+      preview: this.#previewMode ? '1' : '0',
+      ..._params,
+    }
+
+    // convert true/false preview param to '1' / '0'
+    if (_params?.preview === true || _params?.preview === false) {
+      params.preview = _params.preview ? '1' : '0'
+    }
+
+    return new URLSearchParams(params).toString();
+  }
+
+  // rome-ignore lint/suspicious/noExplicitAny: <explanation>
+  async get<T extends string | object> (url: string, params?: Record<string, any>): Promise<T> {
```

Comment:
Could you try using `unknown` rather than `any`? The difference is, if you use `unknown`, TS won't allow you to change it's type, whereas `any` will. `unknown` is seen as a safer type.

Reply:
This causes TS issues. I could refactor to import all `...Params`  types, but need to check impact - e.g. might have impact on the `beforeHook` and `afterHook`. I think it should be possible to update all instances to provide actual expected types.